atomic_reactor/plugins/post_import_image.py

```diff
@@ -83,15 +86,17 @@ def run(self):
             osbs.create_image_stream(self.imagestream, self.docker_image_repo,
                                      **kwargs)
         else:
-            self.log.info("Importing tags for %s", self.imagestream)
-            retry_attempts = 3
-            while True:
-                result = osbs.import_image(self.imagestream, **kwargs)
-                if result:
+            self.log.info("Importing new tags for %s", self.imagestream)
+
+            for attempts in range(self.retry_attempts):
+                new_tags_imported = osbs.import_image(self.imagestream)
+                if new_tags_imported:
                     break
 
-                if retry_attempts > 0:
-                    retry_attempts -= 1
-                    self.log.info("no new tags, will retry after %d seconds",
-                                  self.retry_delay)
-                    sleep(self.retry_delay)
+                self.log.info("no new tags, will retry after %d seconds (%d/%d)",
+                              self.retry_delay, attempts + 1, self.retry_attempts)
+                sleep(self.retry_delay)
+
+            if not new_tags_imported:
+                msg = "Failed to import new tags for %s"
+                raise PluginFailedException(msg % self.imagestream)
```

Comment:
It would be better to raise a `ValueError`, or `RuntimeError`, the atomic-reactor internals will ensure the exception is properly wrapped in `PluginFailedException`.